script.js

```diff
@@ -0,0 +1,93 @@
+let defaultResult = 0;
+
+let currentResult = defaultResult;
+
+let logEntries = [];
+
+function getUserInput() {
+  return parseInt(userInput.value);
+}
+
+// the description & result output of calculation
+
+function createOutput(numberBeforeCalc, operator, calcNumber) {
+  const calcDescription = `${numberBeforeCalc} ${operator} ${calcNumber}`;
+  outputResult(currentResult, calcDescription);
+}
+
+// object inside array
+
+function writeToLog(
+  operationIdentifier,
+  prevResult,
+  numberEnteredByUser,
+  finalResult
+) {
+  const logEntry = {
+    operation: operationIdentifier,
+    PrevResult: prevResult,
+    number: numberEnteredByUser,
+    output: finalResult,
+  };
+
+  logEntries.push(logEntry);
+  console.log(logEntries);
+}
+
+// using if statement instead of seperate functions
+
+function calculationResult(calculationType) {
+  if (
+    (calculationType !== "addition" &&
+      calculationType !== "subtraction" &&
+      calculationType !== "multiplication" &&
+      calculationType !== "division") ||
+    getUserInput() === 0
+  ) {
+    return;
```

Comment:
on this line , what will return do ?

Reply:
➡️ Here, return returns nothing! It is added to quit/cancel function execution. 
➡️ If all of these conditions are met then there is no point in executing the code. so we quit execution to improve response time thereby increasing the performance of JS engine. 
➡️ Hope that's correct 🙂 @anask601

Reply:
Nice @Rehan-Khan-Engineer